Services/SubjectRegistration/ProgressValidation/StudentCsOrCpRegistrationProgress.cs

```diff
@@ -12,6 +12,6 @@
 	public bool DoesRequireCsOrCpValidation { get; init; } = DoesRequireCsOrCpValidation;
 
 	public bool MeetsCriteria =>
-		(DoesRequireCsOrCpValidation)
-		&& (AmountOfDonatedHoursInCsOrCp >= RequiredAmountOfDonatedHoursInCsOrCp);
+		((DoesRequireCsOrCpValidation) && (AmountOfDonatedHoursInCsOrCp >= RequiredAmountOfDonatedHoursInCsOrCp))
+		|| (!DoesRequireCsOrCpValidation);
```

Comment:
Mimochodem tohle se obvykle zapisuje bez té `(DoesRequireCsOrCpValidation) &&` části.
Ale každý to má jinak, jak se ti to bude přirozeněji číst, jestli si řekneš:
1. "podmínka je splněna, nebo podmínka není vyžadována", nebo si řekneš
2. "podmínka je vyžadována a splněna, nebo podmínka není vyžadována".